dotnet/src/Connectors/Connectors.OpenAI/OpenAIMemoryBuilderExtensions.cs

```diff
@@ -33,7 +33,7 @@ public static class OpenAIMemoryBuilderExtensions
         string? modelId = null,
         HttpClient? httpClient = null)
     {
-        return builder.WithTextEmbeddingGeneration((loggerFactory, httpClient) =>
+        return builder.WithTextEmbeddingGeneration((loggerFactory, _) =>
```

Comment:
This solution ignores the HTTP client set up by the `MemoryBuilder.WithHttpClient` method. Consider changing the parameter name for the builder HTTP client to `builderHttpClient` and use it first. If it's not provided, use the one from the `WithAzureOpenAITextEmbeddingGeneration` method. For example:
```C#
public static MemoryBuilder WithAzureOpenAITextEmbeddingGeneration(
    this MemoryBuilder builder,
    string deploymentName,
    string endpoint,
    string apiKey,
    string? modelId = null,
    HttpClient? httpClient = null)
{
    return builder.WithTextEmbeddingGeneration((loggerFactory, builderHttpClient) =>
        new AzureOpenAITextEmbeddingGenerationService(
            deploymentName,
            endpoint,
            apiKey,
            modelId,
            HttpClientProvider.GetHttpClient(builderHttpClient ?? httpClient),
            loggerFactory));
}
```

Reply:
Ok, I can change this. But it's not better to give passed httpClient priority over builderHttpClient?

Reply:
Correct, the HTTP client passed to the `WithAzureOpenAITextEmbeddingGeneration` should take precedence over the builder's one - `HttpClientProvider.GetHttpClient(httpClient ?? builderHttpClient)`.

Reply:
Fixed :)